src/dune/simple_rules.ml

```diff
@@ -35,10 +56,43 @@ let check_filename =
       Path.as_in_build_dir_exn p
     | Dir p -> not_in_dir ~error_loc (Path.to_string p)
 
+type rule_kind =
+  | Alias_only of Alias.Name.t
+  | Alias_with_targets of Alias.Name.t * Path.Build.t
+  | No_alias
+
+let rule_kind ~(rule : Rule.t) ~action =
+  match rule.alias with
+  | None -> No_alias
+  | Some alias -> (
+    match Build.targets action |> Path.Build.Set.choose with
```

Comment:
Is it well-defined what choice `choose` makes? I think we should consistently make the same choice.

Reply:
True, but I can't think of another way to consistently make the same choice. Do you have a suggestion?

Reply:
If the answer is "yes" then I don't have a better suggestion, this is good. If the answer is no, then we should choose (e.g.) the minimum instead.

Reply:
Looks like the answer is "yes", at least in the sense that matters:

```
    (** Return one element of the given set, or raise [Not_found] if
       the set is empty. Which element is chosen is unspecified,
       but equal elements will be chosen for equal sets. *)
```